gluten-data/src/main/java/io/glutenproject/columnarbatch/ArrowColumnarBatches.java

```diff
@@ -169,6 +169,10 @@ public static ColumnarBatch ensureOffloaded(BufferAllocator allocator, ColumnarB
    * will take place if loading is required, which means when the input batch is not loaded yet.
    */
   public static ColumnarBatch ensureLoaded(BufferAllocator allocator, ColumnarBatch batch) {
+    if (batch.numCols() == 0) {
```

Comment:
When `batch.numCols() = 0` the plan should be fall-backed by a rule (https://github.com/oap-project/gluten/blob/20de9e00c3576ebcdb3f90681e6d44f101711385/gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala#L217-L232)

Why the rule didn't work?

Reply:
I find this rule is added in `preOverrides`, but this case occurs in C2R. I guess that's why Gluten C2R has been used.